lua/cfc_http_restrictions/wrap_functions.lua

```diff
@@ -107,6 +107,45 @@ local function wrapPlayURL()
     end
 end
 
+
+local function wrapHTMLPanel(panelName)
+    print( "wrapped SetHTML and OpenURL for "..panelName )
+    funcName = function(functionName)
+        return "_"..panelName.."_"..functionName
+    end
+
+    _G[funcName("SetHTML")] =  _G[funcName("SetHTML")] or vgui.GetControlTable(panelName).SetHTML
+    _G[funcName("OpenURL")] =  _G[funcName("OpenURL")] or vgui.GetControlTable(panelName).OpenURL
+    
+    vgui.GetControlTable(panelName).SetHTML = function( self, html, ... )
+
+        local isAllowed, url = CFCHTTP.isHTMLAllowed( html ) 
+
+        local stack = string.Split( debug.traceback(), "\n" )
+        logRequest( "GET", url, stack[3], isAllowed )
+
+        if not isAllowed then
+            html = [[<h1> BLOCKED </h1>]] 
+        end
+    
+        _G[funcName("SetHTML")]( self, html, ... )
+    end
+    
+    vgui.GetControlTable(panelName).OpenURL = function( self, url, ... )
+        local isAllowed = CFCHTTP.isAllowed( url )
+        local stack = string.Split( debug.traceback(), "\n" )
+        logRequest( "GET", url, stack[3], isAllowed )
+        if not isAllowed then return end
+
+        _G[funcName("OpenURL")]( self, url, ... )
+    end
```

Comment:
Style nitpick - I'm sure you were going to get to it, but thought I'd throw a suggestion in if it made your life easier
```suggestion
    local controlTable = vgui.GetControlTable( panelName )
    
    local setHTML = funcName( "SetHTML" )
    local openURL = funcName( "OpenURL" )

    _G[setHTML] =  _G[setHTML] or controlTable.SetHTML
    _G[openURL] =  _G[openURL] or controlTable.OpenURL
    
    controlTable.SetHTML = function( self, html, ... )
        local isAllowed, url = CFCHTTP.isHTMLAllowed( html ) 

        local stack = string.Split( debug.traceback(), "\n" )
        logRequest( "GET", url, stack[3], isAllowed )

        if not isAllowed then
            html = [[<h1> BLOCKED </h1>]] 
        end
    
        _G[setHTML]( self, html, ... )
    end
    
    controlTable.OpenURL = function( self, url, ... )
        local isAllowed = CFCHTTP.isAllowed( url )

        local stack = string.Split( debug.traceback(), "\n" )
        logRequest( "GET", url, stack[3], isAllowed )

        if not isAllowed then return end

        _G[openURL]( self, url, ... )
    end
```